addons/mail/static/src/model/model_manager.js

```diff
@@ -1036,94 +1043,20 @@ export class ModelManager {
             // Add field accessors.
             for (const field of Model.__fieldList) {
                 Object.defineProperty(Model.prototype, field.fieldName, {
-                    get() {
-                        return field.get(this); // this is bound to record
+                    get() { // this is bound to record
+                        for (const listener of modelManager._listeners) {
+                            modelManager._localIdsObservedByListener.get(listener).add(this.localId);
+                            modelManager._listenersObservingLocalId.get(this.localId).add(listener);
+                            modelManager._listenersObservingFieldOfLocalId.get(this.localId).get(field).add(listener);
```

Comment:
I think the code of observing records and fields would be easier to understand if this PR introduced the concepts of `Observer` and `Observee`.

Observers are components and computed fields that are actively observing records and/or fields.
Observees are records and fields that are actively observed by some observees.

However, I don't think it's worth doing that right now, because I expect it being too hacky with current code.
Components are no models, and there is no proper records for record fields.
It's a bad idea to introduce concepts which mixes model and non-models... And also we don't have proper model combination (~ "Mixin") yet.

This is just comment for future improvement, I don't incite making such big changes right now.
However, if you think you can improve this PR easily with this consideration, let me know ^^